### PR TITLE
Fix warnings and dependencies when using _ALT in test

### PR DESCRIPTION
**Pull request by stevew817**
## Description
Fixing a few issues in the test setup when certain configurations are in use:
* `ecdsa.c` had some unused reference errors when _ALT hooks are in use
* PSA Cipher tests using PKCS7 padding were missing a dependency on PKCS7
* HKDF deriving a maximum-length key qualifies as a RAM-intensive test and should thus be run on targets with sufficient heap.

## Status
**READY**

## Requires Backporting
NO 

## Migrations
NO

### CODE REVIEW

**Review by chris-jones-arm — CHANGES_REQUESTED** [review]
A few changes and additions needed to complete this PR. I also think that this PR is worth backporting to 2.16 although not 2.7 as it is entering end of life and this is non-critical.

File: tests/suites/test_suite_psa_crypto.data [hunk 1]
```diff
@@ -2516,7 +2516,7 @@ derive_key:PSA_ALG_HKDF(PSA_ALG_SHA_256):"0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
 # and not expected to be raised any time soon) is less than the maximum
 # output from HKDF-SHA512 (255*64 = 16320 bytes).
 PSA key derivation: largest possible key
-depends_on:MBEDTLS_SHA512_C
+depends_on:MBEDTLS_SHA512_C:HAVE_RAM_AVAILABLE_128K
 derive_key:PSA_ALG_HKDF(PSA_ALG_SHA_512):"0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b":"000102030405060708090a0b0c":"f0f1f2f3f4f5f6f7f8f9":PSA_KEY_TYPE_RAW_DATA:PSA_MAX_KEY_BITS:PSA_SUCCESS
 
 PSA key derivation: key too large
```
[the remark this answers is not in the document]
Comment (stevew817) [reply tests/suites/test_suite_psa_crypto.data:2522]: Done

File: tests/suites/test_suite_psa_crypto.data [hunk 2]
```diff
@@ -2516,7 +2516,7 @@ derive_key:PSA_ALG_HKDF(PSA_ALG_SHA_256):"0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b
 # and not expected to be raised any time soon) is less than the maximum
 # output from HKDF-SHA512 (255*64 = 16320 bytes).
 PSA key derivation: largest possible key
-depends_on:MBEDTLS_SHA512_C
+depends_on:MBEDTLS_SHA512_C:HAVE_RAM_AVAILABLE_128K
```
[the remark this answers is not in the document]
Comment (stevew817) [reply tests/suites/test_suite_psa_crypto.data:2519]: That was the plan :)

File: tests/suites/test_suite_psa_crypto.data [hunk 3]
```diff
@@ -1241,15 +1241,15 @@ depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
...
-depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_CIPHER_PADDING_PKCS7
 cipher_decrypt:PSA_ALG_CBC_PKCS7:PSA_KEY_TYPE_AES:"2b7e151628aed2a6abf7158809cf4f3c":"2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a":"6279b49d7f7a8dd87b685175d4276e24":"6bc1bee22e409f96e93d7e11739317":PSA_SUCCESS
 
 PSA symmetric decrypt: AES-CBC-PKCS#7, input too short (15 bytes)
-depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_CIPHER_PADDING_PKCS7
 cipher_decrypt:PSA_ALG_CBC_PKCS7:PSA_KEY_TYPE_AES:"2b7e151628aed2a6abf7158809cf4f3c":"2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a":"6bc1bee22e409f96e93d7e11739317":"49e4e66c89a86b67758df89db9ad6955":PSA_ERROR_INVALID_ARGUMENT
 
 PSA symmetric decrypt: AES-CTR, 16 bytes, good
```
[the remark this answers is not in the document]
Comment (stevew817) [reply tests/suites/test_suite_psa_crypto.data:1255]: Done

File: library/ecdsa.c [hunk 4]
```diff
@@ -217,6 +217,9 @@ static void ecdsa_restart_det_free( mbedtls_ecdsa_restart_det_ctx *ctx )
 
 #endif /* MBEDTLS_ECP_RESTARTABLE */
 
+#if defined(MBEDTLS_ECDSA_DETERMINISTIC) || \
+    ! defined(MBEDTLS_ECDSA_SIGN_ALT) || \
+    ! defined(MBEDTLS_ECDSA_VERIFY_ALT)
```
[the remark this answers is not in the document]
Comment (stevew817) [reply library/ecdsa.c]: done

**Comment by stevew817** [comment]
@chris-jones-arm I assume the backport you're requesting only applies to the changes in `ecdsa.c`?

**Comment by chris-jones-arm** [comment]
@stevew817 Sorry for not making that clearer, yes only the changes in `ecdsa.c`.

**Review by chris-jones-arm — DISMISSED** [review]
Thanks for addressing my comments @stevew817, it looks good to me now.

File: tests/suites/test_suite_psa_crypto.data [hunk 5]
```diff
@@ -1161,7 +1161,7 @@ depends_on:MBEDTLS_ARC4_C:MBEDTLS_CIPHER_MODE_CTR
 cipher_setup:PSA_KEY_TYPE_ARC4:"000102030405060708090a0b0c0d0e0f":PSA_ALG_CTR:PSA_ERROR_NOT_SUPPORTED
 
 PSA cipher: bad order function calls
-depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_CIPHER_PADDING_PKCS7
```
Comment (ronald-cron-arm) [inline tests/suites/test_suite_psa_crypto.data:1164]: As the dependency is in the code and not in the test case data it is better to add the dependency on the test case function.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]
I have just one minor comment on this PR.

File: tests/suites/test_suite_psa_crypto.data (hunk 5 above)
Comment (stevew817) [reply tests/suites/test_suite_psa_crypto.data:1164]: That's true for the entire dependency list, no?

Comment (gilles-peskine-arm) [reply tests/suites/test_suite_psa_crypto.data]: Yes (same as https://github.com/ARMmbed/mbedtls/pull/4012 did for `hash_bad_order` and `mac_bad_order`).

**Review by ronald-cron-arm — APPROVED** [review]
Thanks for the extra clean-up in the last commit. LGTM.

**Comment by ronald-cron-arm** [comment]
@chris-jones-arm still ok on your side?

**Review by chris-jones-arm — APPROVED** [review]
Yes it still looks good to me.

**Pull request merged**